src/k8s-engine/wsl.ts

```diff
@@ -1137,73 +1152,106 @@ export default class WSLBackend extends events.EventEmitter implements K8s.Kuber
         this.lastCommandComment = 'Mounting WSL data';
         await this.progressTracker.action(this.lastCommandComment, 100, this.mountData());
         this.lastCommandComment = 'Starting WSL environment';
-        await Promise.all([
+        const installerActions = [
           this.progressTracker.action(this.lastCommandComment, 100, async() => {
             const logPath = await this.wslify(paths.logs);
-            const rotateConf = LOGROTATE_K3S_SCRIPT.replace(/\r/g, '').replace('/var/log', logPath);
 
-            await this.writeFile('/etc/init.d/k3s', SERVICE_SCRIPT_K3S, 0o755);
-            await this.writeFile('/etc/init.d/docker', SERVICE_SCRIPT_DOCKERD, 0o755);
-            await this.writeConf('docker', {
-              WSL_HELPER_BINARY: await this.getWSLHelperPath(),
-              LOG_DIR:           logPath,
-            });
-            await this.writeFile('/etc/logrotate.d/k3s', rotateConf, 0o644);
+            if (enabledK3s) {
+              const rotateConf = LOGROTATE_K3S_SCRIPT.replace(/\r/g, '')
+                .replace('/var/log', logPath);
+
+              await this.writeFile('/etc/init.d/k3s', SERVICE_SCRIPT_K3S, 0o755);
+              await this.writeFile('/etc/logrotate.d/k3s', rotateConf, 0o644);
+            }
+            if (this.#currentContainerEngine === ContainerEngine.CONTAINERD) {
+              await this.execCommand('mkdir', '-p', '/etc/cni/net.d');
+              await this.writeFile('/etc/cni/net.d/10-flannel.conflist', FLANNEL_CONFLIST, 0o644);
+              await this.writeFile('/etc/containerd/config.toml', CONTAINERD_CONFIG, 0o644);
```

Comment:
It may be better to write our custom containerd config to a different path, and add `--config=…` to `containerd_opts` instead.

Reply:
Why?